modules/multisignatures.js

```
@@ -80,158 +80,221 @@ function Multisignatures(cb, scope) {
+	);
 
-		if (
-			multisignatureTransaction.asset.multisignature.signatures ||
-			multisignatureTransaction.signatures.indexOf(transaction.signature) !== -1
-		) {
-			return setImmediate(cb, 'Permission to sign transaction denied');
-		}
+	// Emit events
+	library.network.io.sockets.emit(
+		'multisignatures/signature/change',
+		transaction
+	);
+	library.bus.message('signature', signature, true);
+	return setImmediate(cb);
+};
 
-		// Find public key
-		let verify = false;
-
-		try {
-			for (
-				let i = 0;
-				i < multisignatureTransaction.asset.multisignature.keysgroup.length &&
-				!verify;
-				i++
-			) {
-				const key = multisignatureTransaction.asset.multisignature.keysgroup[
-					i
-				].substring(1);
-				verify = library.logic.transaction.verifySignature(
-					multisignatureTransaction,
-					key,
-					transaction.signature
-				);
+/**
+ * Process signature for transactions that comes from already estabilished multisignature account
+ *
+ * @private
+ * @param {Object} signature - Signature data
+ * @param {string} [signature.publicKey] - Public key of account that created the signature (optional)
+ * @param {string} signature.transactionId - Id of transaction that signature was created for
+ * @param {string} signature.signature - Actual signature
+ * @param {Object} transaction - Corresponding transaction grabbed from transaction pool
+ * @param {function} cb - Callback function
+ * @returns {setImmediateCallback} cb, err
+ */
+__private.processSignatureFromMultisignatureAccount = (
+	signature,
+	transaction,
+	cb
+) => {
+	// Get sender account of correscponding transaction
+	modules.accounts.getAccount(
+		{ address: transaction.senderId },
+		(err, sender) => {
+			if (err || !sender) {
+				const message = 'Unable to process signature, account not found';
+				library.logger.error(message, { signature, transaction, error: err });
+				return setImmediate(cb, message);
```

Please return an error object instead `new Error(message)`